Update Configuration Documentation

jhsimpson opened the pull request:
This PR updates the readme files created in #845, adding more explanatory text.  

The gunicorn configuration variables are described for the dashboard, and the dashboard sample systemd config file is renamed to match the name of the file that is installed.

jhsimpson committed:
suggested edits to the Install/readme.md

sevein commented on src/MCPClient/install/README.md:
```diff
@@ -24,18 +38,35 @@ configuration system coerces the value to the types supported:
+  - False (disabled): `"0"`, `"no"`, `"false"` or `"off"`
 
-Certain environment strings are mandatory, i.e. they don't have defaults and
+Certain environment strings aremandatory, i.e. they don't have defaults and
```
Typo?

sevein commented on src/dashboard/install/archivematica-dashboard.service:
```diff
@@ -1,4 +1,4 @@
-description "Archivematica Storage Service"
+description "Archivematica Dashboard"
```
Good catch. But I don't understand why you've renamed this file, it's an upstart config file and their standard extension is `.conf`.

jhsimpson replied:
It's a systemd service file, default extension is .service.   It is described in the readme.md as a systemd config file, not upstart, and it is being used as such by am-packbuild and the archivematica ansible role, isn't it?

sevein replied:
systemd units have a different format AFAIK. It's possible that am-packbuild provides its own systemd files. @scollazo 🏓

jhsimpson replied:
ok, I see the format of this file is not systemd.  However the readme refers to it as a systemd example - either the syntax of this file should be updated to provide valid systemd synax, or the name of the file should be reverted to .conf and the readme updated to refer to it as an upstart example.

I think it should be systemd , so I will provide a new commit that updates the contents of the file.

sevein commented on src/dashboard/install/README.md:
```diff
@@ -150,28 +206,79 @@ This is the full list of environment strings supported:
+Archivematica 1.6.1 and earlier releases are configured by default to log to 
+subdirectories in /var/log/archivematica, such as 
+/var/log/archivematica/dashboard/dashboard.debug.log. Starting with 
+Archivematica 1.7.0, configuration defaults to using stdout and stderr
```
In this change block, why do you delete the backticks?
Also you've added trailing whitespaces to three of the four lines.

sevein commented on src/dashboard/install/README.md:
```diff
@@ -34,7 +49,48 @@ Please be aware that Archivematica supports different types of distributions
+## Configuration files
+
+The Dashboard will look for a configuration file in one location:
+-`/etc/archivematica/archivematicaCommon/dbsettings`
```
This list isn't formatted properly. I can't tell if it's the lack of indentation or the fact that it's not separated from the paragraph like we do in other places in this document.

sevein commented on src/dashboard/install/README.md:
```diff
@@ -150,28 +206,79 @@ This is the full list of environment strings supported:
 
-## Other configuration files
+- **`AM_GUNICORN_USER`**:
+    - **Description:** OS user for gunicorn worker processes to run as.  See [USER](http://docs.gunicorn.org/en/stable/settings.html#user)
```
All the _Description_ fields under _Gunicorn variables_ are missing the period at the end of the sentence.

sevein commented on src/MCPServer/install/README.md:
```diff
@@ -35,7 +49,23 @@ Please be aware that Archivematica supports different types of distributions
+
+MCPClient will look for a configuration file in one of two locations:
+-`/etc/archivematica/archivematicaCommon/dbsettings` 
+ -`/etc/archivematica/MCPServer/serverConfig.conf` 
```
Same formatting issue in this list.

sevein reviewed (changes requested):
This is looking very good, @jhsimpson. I've added some suggestions but it looks good to me.

sevein commented on src/dashboard/install/README.md:
```diff
@@ -1,18 +1,33 @@
 - [Environment variables](#environment-variables)
-- [Other configuration files](#other-configuration-files)
+- [Configuration files](#configuration-files)
+- [Parameter list](#parameter-list)
```
Should we also add the subsections to the TOC?

```
- [Introduction](#introduction)
- [Environment variables](#environment-variables)
- [Configuration files](#configuration-files)
- [Parameter list](#parameter-list)
  - [Application variables](#application-variables)
  - [Gunicorn variables](#gunicorn-variables)
- [Logging configuration](#logging-configuration)
```

sevein commented:
I've updated my PR in SS to be more aligned with the work you've done here, @jhsimpson.
Here it is: https://github.com/artefactual/archivematica/pull/856/files.
https://github.com/artefactual/archivematica-storage-service/pull/294

jhsimpson commented:
Other than one comment about upstart vs systemd extension, I have incorporated all the suggested edits @sevein .  ready for re-review.

sevein reviewed (changes requested):
LGTM, @jhsimpson. Only that bit about the service unit being an upstart configuration file. am-packbuild seems to provide systemd configurations but uses the upstart configurations in this repo.

jhsimpson committed:
Add systemd example config for dashboard

this commit clarifies the difference between
the systemd and upstart example configs
in the install readme for the dashboard.

jhsimpson commented:
ping @sevein does the most recent commit to this branch address your last point?

sevein commented on src/dashboard/install/README.md:
```diff
@@ -77,10 +77,14 @@ This directory contains example configuration files for these services:
 
+- [`archivematica-dashboard.conf`](./archivematica-dashboard.conf) - upstart
+   config sample, for use on Ubuntu 14.04 where systemd is not available. The
+   default location for this file is /etc/init/archivematica-dashboard.conf
```
End of sentence missing period.

jhsimpson committed:
fixing typo

jhsimpson commented:
I fixed the typo you spotted @sevein

jhsimpson commented:
rebased to include more recent commits in stable/1.7.x

jhsimpson commented:
cherry-picked to qa/1.x as:
196138243db72f0ed6b89e7e921acb4a3c219cf4
cb9d11ef8f94b4b4614b8a02d505b4847774c6da
0e01aa2ef85290325ccc2474efa45e4066db836d

sevein commented:
This should have been squashed!
The branch hasn't been deleted btw.
